src/v/cloud_storage/topic_manifest.h

```diff
@@ -19,8 +19,15 @@
 
 namespace cloud_storage {
 
+namespace testing {
+void topic_manifest_serialize_v1_json(std::ostream&, const topic_manifest& m);
+}
+
```

Comment:
@nvartolomei i think you can get rid of the testing namespace completely in this non-test context using friend class. see here for example `src/v/storage/segment_appender.h:    friend struct segment_appender_test_accessor`.

Reply:
Is there a big difference between forward declaring a struct vs a function? Btw, we enclose these forward declarations in a testing/testing_details namespace in other places too.